xarray/plot/dataset_plot.py

```diff
@@ -450,3 +451,53 @@ def scatter(ds, x, y, ax, **kwargs):
         )
 
     return primitive
+
+
+def _attach_to_plot_class(plotfunc):
+    """Set the function to the plot class and add common docstring."""
+    # Build on the original docstring:
+    original_doc = getattr(_PlotMethods, plotfunc.__name__, None)
+    commondoc = original_doc.__doc__
+    if commondoc is not None:
+        doc_warning = (
+            f"This docstring was copied from xr.DataArray.plot.{original_doc.__name__}."
+            " Some inconsistencies may exist."
+        )
+        # Add indentation so it matches the original doc:
+        commondoc = f"\n\n    {doc_warning}\n\n    {commondoc}"
+    else:
+        commondoc = ""
+    plotfunc.__doc__ = (
+        f"    {plotfunc.__doc__}\n\n"
+        "    The y DataArray will be used as base,"
+        "    any other variables are added as coords.\n\n"
+        f"{commondoc}"
+    )
```

Comment:
Adding an explanation here how the dataset becomes a dataarray should help making sense of the autogenerated docs I think.

Reply:
Does the `_dsplot` decorator not work for this case? It would help minimize the changes.

Reply:
Now that the line plot comes from the dataarray side you don't have to do all the additional things `_dsplot` is doing. All that's really needed then is to attach to the plot class and then do this tweak to the docs so that they make more sense.

My end goal here is to move all the dataset specific plots to the dataarray side and use this thin wrapper for dataset plotting. Back when I started this PR in january there was only `scatter` so it made more sense back then.

Reply:
Thanks @Illviljan Can you add this to the docstring of `_attach_to_plot_class`

Let's also see what @mathause thinks...